### PR TITLE
Added general dielectric conductor Fresnel.

### PR DESCRIPTION
**Pull request by est77**
And use it in the metal brdf. Removed unused code.

### CODE REVIEW

File: src/appleseed/foundation/math/fresnel.h [hunk 1]
```diff
@@ -140,14 +140,27 @@ void normal_reflectance_dielectric(
...
+
 
 // Compute the Fresnel reflectance for a conductor for unpolarized light.
-// This approximation assumes an air - conductor interface.
 template <typename SpectrumType, typename T>
 void fresnel_reflectance_conductor(
     SpectrumType&           reflectance,
-    const SpectrumType&     n,                  // refractive index.
-    const SpectrumType&     k,                  // extinction coefficient.
+    const SpectrumType&     nt,                  // conductor refractive index.
```
Comment (dictoon) [inline src/appleseed/foundation/math/fresnel.h]: Remove period at the end of inline comments.

File: src/appleseed/foundation/math/fresnel.h [hunk 2]
```diff
@@ -435,50 +448,70 @@ void normal_reflectance_dielectric(
...
 
-    const SpectrumType n2 = n * n;
-    const SpectrumType k2 = k * k;
-    const SpectrumType n2_plus_k2 = n2 + k2;
-    const SpectrumType n_2cos_theta = n * ValueType(2.0) * static_cast<ValueType>(cos_theta);
+    SpectrumType a2plusb2 = t0 * t0;
+    SpectrumType tmp = nt2 * kt2;
+    tmp *= T(4.0);
+    a2plusb2 += tmp;
+    a2plusb2 = sqrt(a2plusb2);
```
Comment (dictoon) [inline src/appleseed/foundation/math/fresnel.h]: `std::sqrt()` (fix everywhere).

Comment (est77) [inline src/appleseed/foundation/math/fresnel.h]: I want to call sqrt(Spectrum), not std::sqrt().

Comment (est77) [inline src/appleseed/foundation/math/fresnel.h]: Yes, but it would be incorrect to call this method with SpectrumType = float or double in any case..

**Comment by est77** [comment]
Fixed comments and rebased.

**Pull request merged**